src/openforms/products/utils/entry.py

```diff
@@ -0,0 +1,26 @@
+import json
+
+from .factory import FieldFactory
+
+
+def convert_json_schema_to_py(json_schema: dict) -> dict:
+    """
+    reads json schema in json format and return dict of fields objects
+    """
+    final_result = {"components": []}
+    properties = json_schema.get("properties", {})
+    for key, content in properties.items():
+        type_flag = content.get("type")
+        if type_flag != "object":
+            # check if attr "required" exists
+            if json_schema.get("required") is not None:
+                required = key in json_schema.get("required")
+            else:
+                required = None
+            single_field = FieldFactory.create(key, required=required, content=content)
+        else:
+            single_field = FieldFactory.create(key, content=content)
```

Comment:
I don't see why object is special (in this function). I even think an object can be required, which now cannot be the case. The code can be easier (please doublecheck!)

```suggestion
        required = key in json_schema.get("required", [])
        single_field = FieldFactory.create(key, required=required, content=content)
```

Reply:
Thought about it. Looks  having a fieldset as "required" not logic; some single fields in the fieldset could. But I'll explore it further.

Reply:
well, whether you do something with it in the FieldSet is upto the FieldSet class - not this function.